nampower/main.cpp

```diff
@@ -97,12 +97,12 @@ void BeginCast(DWORD currentTime, std::uint32_t castTime, int spellId)
 
     gLastCast = currentTime;
 #endif
-
-    if (castTime)
-    {
-        void(*signalEvent)(std::uint32_t, const char *, ...) = reinterpret_cast<decltype(signalEvent)>(Offsets::SignalEventParam);
-        signalEvent(game::Events::SPELLCAST_START, "%s%d", game::GetSpellName(spellId), castTime);
-    }
+    //JT: Use the notification from server to activate cast bar - it is needed to have HealComm work.
+    //if (castTime)
+    //{
+    //    void(*signalEvent)(std::uint32_t, const char *, ...) = reinterpret_cast<decltype(signalEvent)>(Offsets::SignalEventParam);
+    //    signalEvent(game::Events::SPELLCAST_START, "%s%d", game::GetSpellName(spellId), castTime);
+    //}
 }
```

Comment:
This change seems suspicious to me.  If we let the server activate the cast bar it will be delayed, which means the ending will also be delayed.  Surely there is another way to appease healcomm?

Reply:
This is purely delaying the castbar client side. The end is not delayed - instead the castbar simply does not fill up 100% before it disappears and the spell is cast. This is what I came up with to fix healcomm and it works like a charm. I tried to explain why I did this in the msg under point 3 and how healcomm works. At least 1 raiding guild on TurtleWoW has fully adopted my version of nampower and tested it for over a month before I created the pull request.